guides/admin/inotify-watch-limits.md

```diff
@@ -0,0 +1,266 @@
+- `fs.inotify.max_user_watches`, which defines the maximum number of files and
+  folders that programs can monitor for changes.
+
+For additional detail regarding the `inotify` system, please see
+[inotify(7)](https://man7.org/linux/man-pages/man7/inotify.7.html).
+
+To see the values for these settings applicable to your environment, run the
+following commands:
+
+```console
+$ sysctl fs.inotify.{max_queued_events,max_user_instances,max_user_watches}
+fs.inotify.max_queued_events = 16384
+fs.inotify.max_user_instances = 128
+fs.inotify.max_user_watches = 8192
+```
+
+Because these settings are not namespace-aware, the values will be the same
+regardless of whether you run the commands on the host system or inside a
+container running on that host.
+
+### Identify inotify consumers
+
+To identify the programs consuming `inotify` watches, you can use a script that
+summarizes the information available in the `/proc` filesystem, such as
+[`inotify-consumers`]. This script will show the names of programs along with
+the number of `inotify` watches registered with the kernel:
+
+```console
+$ ./inotify-consumers
+   INOTIFY
+   WATCHER
+    COUNT     PID USER     COMMAND
+--------------------------------------
+     269   254560 coder    /opt/coder/code-server/lib/node /opt/coder/code-server/lib/vscode/out/bootstrap-fork --type=watcherService
+       5     1722 coder    /opt/coder/code-server/lib/node /opt/coder/code-server/lib/vscode/out/vs/server/fork
+       2   254538 coder    /opt/coder/code-server/lib/node /opt/coder/code-server/lib/vscode/out/bootstrap-fork --type=extensionHost
+       2     1507 coder    gpg-agent --homedir /home/coder/.gnupg --use-standard-socket --daemon
+
+     278  WATCHERS TOTAL COUNT
+```
+
+Please note that this is a third-party script published by an individual who is
+not affiliated with Coder, and as such, we cannot provide a warranty or support
+for its usage.
+
+[`inotify-consumers`]: https://github.com/fatso83/dotfiles/blob/master/utils/scripts/inotify-consumers
+
+To see the specific files and directories that the tools track for changes, you
+can use `strace` to monitor invocations of the `inotify_add_watch` system call:
+
+```console
+$ strace --follow-forks --trace='inotify_add_watch' inotifywait --quiet test
+inotify_add_watch(3, "test", IN_ACCESS|IN_MODIFY|IN_ATTRIB|IN_CLOSE_WRITE|IN_CLOSE_NOWRITE|IN_OPEN|IN_MOVED_FROM|IN_MOVED_TO|IN_CREATE|IN_DELETE|IN_DELETE_SELF|IN_MOVE_SELF) = 1
+```
+
+This shows that the `inotifywait` command is listening for notifications related
+to the `test` file.
+
+## Resolution
+
+If you encounter the file watcher limit, there are two potential resolutions:
+either reduce the number of file watcher registrations, or increase the maximum
+file watcher limit. We recommend attempting to reduce the file watcher
+registrations first, because increasing the number of file watches may result
+in high processor utilization.
```

Comment:
Would recommend using:

```md
1. Reduce the number of file watcher registrations
1. Increase the maximum file watcher limit

We recommend...
```

structure so as to call out the information

Reply:
nice, done